didland/src/lib.rs

```diff
@@ -43,303 +97,504 @@ fn init(path: &str) -> Result<String, std::io::Error> {
 
-    // 3. Serialize did-document to json
+    use did_key::DIDCore;
+    let did_doc = self_didkey.get_did_document(did_key::CONFIG_LD_PUBLIC);
     let did_doc = serde_json::to_string_pretty(&did_doc).unwrap();
+
     Ok(format!("{}", did_doc))
 }
 
 
-fn did(path: &str) -> Result<String, std::io::Error> {
-    use did_key::DIDCore;
-
-    // 1. Read jwk from file
-    let jwk = std::fs::read(didkey_jwk_path(path))?;
-    let jwkstr = String::from_utf8(jwk).unwrap();
-    let (public,_) = jwkstr_to_publicprivatebytes(&jwkstr);
-
-    // 2. Transform public key to a did-document
-    let keypair = did_key::from_existing_key::<did_key::Ed25519KeyPair>(&public, None);
-    let did_doc = keypair.get_did_document(did_key::CONFIG_LD_PUBLIC);
+fn did() -> Result<String, std::io::Error> {
+    let self_didkey = get_self_didkey();
 
-    let did = did_doc.id;
+    use did_key::DIDCore;
+    let diddoc = self_didkey.get_did_document(did_key::CONFIG_LD_PUBLIC);
+    let did = diddoc.id;
 
-    // 3. Print did
     Ok(format!("{}", did))
 }
 
-fn connect(path: &str, name: &str, did: &str) -> Result<String, std::io::Error> {
+fn connect(to_did_name: &str, did: &str) -> Result<String, std::io::Error> {
     use std::io::Write;
 
-    // 2. Create 'name' -> 'did' mapping
-    let mut file = std::fs::File::create(name_path(path, name)).unwrap();
-    file.write(did.as_bytes()).unwrap();
+    // 1. Create 'name' -> 'did'-mapping
+    let mut file = std::fs::File::create(name_path(to_did_name))?;
+    file.write(did.as_bytes())?;
 
-    // 3. Create 'did' to 'name' mapping
-    let mut file = std::fs::File::create(did_path(path, did)).unwrap();
-    file.write(name.as_bytes()).unwrap();
+    // 2. Create 'did' -> 'name'-mapping
+    let mut file = std::fs::File::create(did_path(did))?;
+    file.write(to_did_name.as_bytes())?;
 
-    Ok(format!("{}\n{}", name_path(path, name), did_path(path, did)))
+    Ok(format!("{}\n{}", name_path(to_did_name), did_path(did)))
 }
 
 
-fn write(path: &str, name: &str, message: &str) -> Result<String, std::io::Error> {
-    use did_key::KeyMaterial;
+fn write(to_did_name: &str, message: &str) -> Result<String, std::io::Error> {
     use std::io::Write;
 
-    // 1. Read from-key
-    let jwk = std::fs::read(didkey_jwk_path(path))?;
-    let jwkstr = String::from_utf8(jwk).unwrap();
-    let (_, private) = jwkstr_to_publicprivatebytes(&jwkstr);
-    let from_key = did_key::Ed25519KeyPair::from_seed(&private);
+    // 1. Get did:keys
+    let from_key = get_self_didkey();
+    let to_key = get_other_didkey(to_did_name);
 
-    // 2. Read to-key
-    let to_did = std::fs::read_to_string(name_path(path, name)).unwrap();
-    let to_key = did_key::resolve(&to_did).unwrap();
-    let to_key = did_key::Ed25519KeyPair::from_public_key(&to_key.public_key_bytes());
+    // 2. Encrypt message with from_key, to keep message history in local file
+    let dcem = encrypt_didcomm(&from_key, &from_key, message);
```

Comment:
This method is using the v2.0 lib? How far away from the spec are you in the way you send things? Is this worth mentioning in the paper?